### ISSUE TITLE
Mapping resource with different name than the controller name

### ISSUE THREAD

**Issue by janslifka** [cakephp/cakephp#5895]
It would be useful to have an option of choosing different url name when mapping resource.

```
$routes->resources('Members', [
    'urlName' => 'users'
]);
```

I think that this change shouldn't be hard to implement. I tried to modify RouterBuilder::resources method. I replaced:

```
$urlName = Inflector::underscore($name);
```

with this:

```
if (!empty($options['urlName'])) {
    $urlName = $options['urlName'];
} else {
    $urlName = Inflector::underscore($name);
}
```

Seems like it works fine. I'm not sure if this could break something else though.

**Comment by ADmad** [comment]
Also it forces underscored inflection which would create inconsistency in url parts if some is using DashedRoute as default route class.

**Comment by markstory** [comment]
Seems like a reasonable idea if we can figure out how to handle inflections.

**Comment by markstory** [comment]
Being able to change the inflection to dashed was implemented in #5934

**Comment by markstory** [comment]
This feature was implemented as the `path` option in 3.5.0
